app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb

```diff
@@ -1,31 +1,53 @@
 module Katello
   module Concerns
-    module Api
-      module V2
-        module RegistrationCommandsControllerExtensions
-          module ApipieExtensions
-            extend ::Apipie::DSL::Concern
-
-            update_api(:create) do
-              param :registration_command, Hash do
-                param :activation_key, String, desc: N_('Activation key(s) for Subscription Manager. Required for CentOS and Red Hat Enterprise Linux. Multiple keys add separated by comma, example: key1,key2,key3.')
-              end
+    module Api::V2::RegistrationCommandsControllerExtensions
+      extend ActiveSupport::Concern
+
+      module Overrides
+        def registration_args
+          args = super
+
+          if args['activation_key'].present?
+            args['activation_keys'] = args['activation_key']
```

Comment:
So if I pass both `activation_key` and `activation_keys` it looks like the former wins by overwriting the latter?

How about starting this off like:

```
args['activation_keys'] ||= []

# parse activation_key param

args['activation_keys'] << parsed_activation_key_param
```

That should prevent the need to check for String and Array which are usually code smells in Ruby